File: frontend/event-people-dashboard/src/types/zone.ts
export interface ZoneResponse {
  id: string;
  name: string;
  description: string;
  capacity: number | null;
  eventId: string;
  createdAt: string;
  boundaryGeoJson: string | null;
  color: string;
  allowedRoles: string;
  accessTags: string;
}

export interface CreateZoneRequest {
  name: string;
  description: string;
  capacity?: number;
  boundaryGeoJson?: string | null;
  color?: string;
  allowedRoles?: string;
  accessTags?: string;
}

File: backend/event-people-manager-api/src/main/java/com/event/peoplemanager/domain/entity/Zone.java
package com.event.peoplemanager.domain.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.ZonedDateTime;
import java.util.UUID;

@Entity
@Table(name = "zones")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Zone {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    private String description;

    private Integer capacity;

    @Column(name = "boundary_geojson")
    private String boundaryGeoJson;

    @Builder.Default
    @Column(name = "color")
    private String color = "#3b82f6";

    @Builder.Default
    @Column(name = "allowed_roles")
    private String allowedRoles = "";

    @Builder.Default
    @Column(name = "access_tags")
    private String accessTags = "";

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "event_id")
    private Event event;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;
}
